src/axom/primal/operators/detail/evaluate_integral_impl.hpp

```diff
@@ -0,0 +1,197 @@
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a double
+ * \param [in] quad the mfem integration rule containing nodes and weights
+ * \return the value of the integral
+ */
+inline double evaluate_line_integral_component(
+  const primal::BezierCurve<double, 2>& c,
+  std::function<double(Point2D)> integrand,
+  const mfem::IntegrationRule* quad)
+{
+  // Store intermediate values
+  Point2D x_q;
+  Vector2D dx_q;
+
+  // Store/compute quadrature result
+  double full_quadrature = 0.0;
+  for(int q = 0; q < quad->GetNPoints(); q++)
+  {
+    // Get intermediate quadrature point
+    //  at which to evaluate tangent vector
+    x_q = c.evaluate(quad->IntPoint(q).x);
+    dx_q = detail::temp_dt(c, quad->IntPoint(q).x);
+
+    full_quadrature += quad->IntPoint(q).weight * integrand(x_q) * dx_q.norm();
+  }
+
+  return full_quadrature;
+}
+
+/*!
+ * \brief Evaluate a vector field line integral on a single Bezier curve.
+ *
+ * Evaluate the vector field line integral with MFEM integration rule
+ *
+ * \param [in] c the Bezier curve object
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a 2D axom vector object
+ * \param [in] quad the mfem integration rule containing nodes and weights
+ * \return the value of the integral
+ */
+inline double evaluate_line_integral_component(
+  const primal::BezierCurve<double, 2>& c,
+  std::function<Vector2D(Point2D)> vec_field,
+  const mfem::IntegrationRule* quad)
+{
+  // Store intermediate values.
+  Point2D x_q;
+  Vector2D dx_q, func_val;
+
+  // Store/compute quadrature result
+  double full_quadrature = 0.0;
+  for(int q = 0; q < quad->GetNPoints(); q++)
+  {
+    // Get intermediate quadrature point
+    //  on which to evaluate dot product
+    x_q = c.evaluate(quad->IntPoint(q).x);
+    dx_q = detail::temp_dt(c, quad->IntPoint(q).x);
+    func_val = vec_field(x_q);
+
+    full_quadrature +=
+      quad->IntPoint(q).weight * Vector2D::dot_product(func_val, dx_q);
+  }
+
+  return full_quadrature;
+}
+
+/*!
+ * \brief Evaluate the integral across a single component of the curved polygon.
+ *
+ * Uses a Spectral Mesh-Free Quadrature derived from Green's theorem, evaluating
+ * the area integral as a line integral of the antiderivative over the curve.
+ *
+ * \param [in] cs the array of Bezier curve objects that bound the region
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a double
+ * \param [in] The lower bound of integration for the antiderivatives
+ * \param [in] quad_Q the quadrature rule for the line integral
+ * \param [in] quad_P the quadrature rule for the antiderivative
+ * \return the value of the integral, which is mathematically meaningless.
+ */
+template <class Lambda>
+double evaluate_area_integral_component(const primal::BezierCurve<double, 2>& c,
```

Comment:
Aside: We might want to generalize the BezierCurve type to `CoordType`s other than `double`.  Should be fine for now though.